compiler/tests-compiler/gh1007.ml

```diff
@@ -158,8 +158,9 @@ let ()  = M.myfun M.x
      var x$0 = x;
      for(;;){
       if(! x$0) return 0;
+      var next = x$0[1], cmp = runtime.caml_int_compare;
+      let cmp$0 = cmp;
```

Comment:
I don't understand this change.

Reply:
Initially, we have:
```
 1 v645 = fun(v647, v646){5 (v646, v647)}

==== 5 (v648, v649) ====
 1 v650 = "caml_int_compare"(v649, v648)
   return v650
```
On master, the function is inlined.
With this PR, the parameters of the initial continuation are eliminated right away (`fun(v647, v646){5 ()}`, so it is turned into an alias:
```
 1 v645 = alias caml_int_compare
```
Then, since the variable is defined in a loop and used in a function, a let binding is introduced.

Reply:
Maybe we should only introduce aliases once after all optimizations passes

Reply:
I've pushed a commit to only emit aliases at the end